README.md

```diff
@@ -34,6 +34,7 @@ This library enables you to manage Artifactory resources such as users, groups,
     + [Deploy an artifact](#deploy-an-artifact)
     + [Deploy an artifact with properties](#deploy-an-artifact-with-properties)
     + [Deploy an artifact with checksums](#deploy-an-artifact-with-checksums)
+    + [Deploy an artifact by checksums](#deploy-an-artifact-by-checksums)
```

Comment:
I don't get why you want to create a new deploy by checksum instead of just editing the existing one. This would lead to confusion to have two ways to perform the same action

Reply:
I understand that there are two ways to deploy artifacts using checksums:

1.  Deploy by checksum: Based on the provided checksums, Artifactory checks if the artifact exists in Artifactory and if so, the artifact is virtually copied to the target location; otherwise no new files are deployed. Ref https://jfrog.com/help/r/jfrog-rest-apis/deploy-artifact-by-checksum
2. Deploy with checksum (less documented than 1): the client deploys the checksums as header with the file (unlike 1, you can deploy new artifacts in Artifactory). Ref https://jfrog.com/help/r/what-are-client-checksum-server-checksum-and-checksum-policy-in-local-repositories/1.-verify-against-client-checksums-this-is-the-default-checksum-policy-for-local-repositories , and https://jfrog.com/help/r/artifactory-how-to-identify-and-fix-all-artifacts-with-missing-client-checksums/artifactory-how-to-identify-and-fix-all-artifacts-with-missing-client-checksums
As I see it, although both actions involve checksums, they both have different results that should be considered as separated methods.
I reckong this sounds a bit complicated, but I'll do my best to implement what is best for users.

Reply:
The first link you shared does not work.

> Deploy with checksum (less documented than 1): the client deploys the checksums as header with the file (unlike 1, you can deploy new artifacts in Artifactory)

Ok, we can support it, but I think it's better to refactor the existing `deploy` method instead of creating a new function. That might confuse the users of the library. Maybe you can do something like this:
```
If checksum_enabled = False:
    headers={
                            "X-Checksum-Deploy": "false",
                            "X-Checksum-Sha1": artifact_check_sums.sha1,
                            "X-Checksum-Sha256": artifact_check_sums.sha256,
                            "X-Checksum": artifact_check_sums.md5,
                        },
```

This way, you will deploy your checksums with the file. But what would happen if we deploy using method 2, then method 1 ("X-Checksum-Deploy": "true")?